hack/tools/release/notes/main.go

```diff
@@ -441,121 +127,85 @@ func commandExists(cmd string) bool {
 	return err == nil
 }
 
-// releaseNoteEntryResult is the result of processing a PR to create a release note item.
-// Used to aggregate the line item and error when processing concurrently.
-type releaseNoteEntryResult struct {
-	prEntry *releaseNoteEntry
-	err     error
-}
+func validateConfig(config *notesCmdConfig) error {
+	if config.fromRef == "" && config.newTag == "" {
+		return errors.New("at least one of --from or --release need to be set")
+	}
 
-// releaseNoteEntry represents a line item in the release notes.
-type releaseNoteEntry struct {
-	title    string
-	section  string
-	prNumber string
-}
+	if config.branch == "" && config.newTag == "" {
+		return errors.New("at least one of --branch or --release need to be set")
+	}
 
-// removePrefixes removes the specified prefixes from the title.
-func removePrefixes(title string, prefixes []string) string {
-	entryWithoutTag := title
-	for _, prefix := range prefixes {
-		entryWithoutTag = strings.TrimLeft(strings.TrimPrefix(entryWithoutTag, prefix), " ")
+	if config.fromRef != "" {
+		if err := validateRef(config.fromRef); err != nil {
+			return err
+		}
 	}
 
-	return entryWithoutTag
-}
+	if config.toRef != "" {
+		if err := validateRef(config.toRef); err != nil {
+			return err
+		}
+	}
 
-// trimAreaFromTitle removes the prefixed area from title to avoid duplication.
-func trimAreaFromTitle(title, area string) string {
-	titleWithoutArea := title
-	pattern := `(?i)^` + regexp.QuoteMeta(area+":")
-	re := regexp.MustCompile(pattern)
-	titleWithoutArea = re.ReplaceAllString(titleWithoutArea, "")
-	titleWithoutArea = strings.TrimSpace(titleWithoutArea)
-	return titleWithoutArea
+	return nil
 }
 
-func capitalize(str string) string {
-	return strings.ToUpper(string(str[0])) + str[1:]
-}
+// computeConfigDefaults calculates the value the non specified configuration fields
+// based on the set fields.
+func computeConfigDefaults(config *notesCmdConfig) error {
```

Comment:
I tried to run this tool for CR with the following args:
```
-from tags/v0.16.0 -branch main -release v0.17.0 -repository kubernetes-sigs/controller-runtime
```

This func returns in l.158. Then l.182 is never executed and it eventually panics in parseRef

```go
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
main.parseRef({0x0, 0x0})
        /Users/buringerst/code/src/sigs.k8s.io/cluster-api/hack/tools/release/notes/ref.go:43 +0xbc
main.(*notesCmd).run(0x1400004e050)
        /Users/buringerst/code/src/sigs.k8s.io/cluster-api/hack/tools/release/notes/main.go:101 +0x124
main.main()
        /Users/buringerst/code/src/sigs.k8s.io/cluster-api/hack/tools/release/notes/main.go:41 +0x28
Exiting.


```

Reply:
Otherwise it works great!

Reply:
You are 100% right
I'll get it fixed

That said, you don't need to specify `-from` in this case. Specifying `-release` is enough for the program to infer the rest. You do need to specify the branch bc it seems like in `controller-runtime` minor releases are cut directly from `main`. That would work even without the fix.

```sh
./bin/notes --branch main --release v0.17.0 --repository kubernetes-sigs/controller-runtime --prefix-area-label=false --add-kubernetes-version-support=false
```

Reply:
Thx, yup sounds reasonable. I just started with adding --from without thinking much about it and then I added what was needed :)

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/9958